compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -1343,25 +1366,21 @@ where
     )
 }
 
-fn if_expr<'a, P1, P2>(
-    expr_parser: P1,
-    expr_no_constructors: P2,
-) -> impl NoirParser<ExpressionKind> + 'a
+fn if_expr<'a, P, S>(expr_no_constructors: P, statement: S) -> impl NoirParser<ExpressionKind> + 'a
 where
-    P1: ExprParser + 'a,
-    P2: ExprParser + 'a,
+    P: ExprParser + 'a,
+    S: NoirParser<Statement> + 'a,
 {
     recursive(|if_parser| {
-        let if_block = block_expr(expr_parser.clone());
+        let if_block = block_expr(statement.clone());
```

Comment:
Oh didn't see this -- why was this changed?

Reply:
The `statement` parser is now recursive. It was surprising it wasn't recursive until this point. For loops are the only statement type now to have blocks in them. The others only have expressions with blocks - for which they can just use the already recursive expression parser.

The fact that the statement parser is now recursive is why I had to change so many other parsers just to pass it around now. This is my main annoyance with chumsky.

Reply:
oof